wpimath/src/main/java/edu/wpi/first/math/geometry/Pose2d.java

```diff
@@ -17,10 +23,22 @@
 /** Represents a 2D pose containing translational and rotational elements. */
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonAutoDetect(getterVisibility = JsonAutoDetect.Visibility.NONE)
-public class Pose2d implements Interpolatable<Pose2d> {
+public class Pose2d
+    implements Interpolatable<Pose2d>, ProtobufObject<ProtobufPose2d>, StructObject {
   private final Translation2d m_translation;
   private final Rotation2d m_rotation;
 
+  public static final String kTypeStringProto = "proto:Pose2d";
+  public static final String kTypeStringStruct = "struct:Pose2d";
+  public static final int kStructSize = Translation2d.kStructSize + Rotation2d.kStructSize;
+
+  static {
```

Comment:
Long-term - is it necessary for `Pose2d` to implement both struct and protobuf serialization?

Reply:
Yes.  We'll want to do that with all the fixed size data types.  The reason is to make them both fast and efficient to send as themselves (e.g. sending just a `Pose2d[]` is quite common) and support sending them in higher level (including vendor/team) protobufs.